### PR TITLE
Add gruvbox base theme and gruvbox green color theme

### LINKED ISSUE #3987: [Feature Request]: Add Gruvbox theme.
### Guidelines

- [X] I have searched the issue tracker for [open](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aopen+is%3Aissue) and [closed](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+is%3Aclosed) issues that are similar to the feature request I want to file, without success.
- [X] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the feature request I want to file, without success.
- [X] This issue contains only one feature request.

### Problem Description

Add the gruvbox dark and gruvbox light themes as an option for selectable themes.

### Proposed Solution

Add both gruvbox dark and gruvbox light themes.

### Alternatives Considered

N/A

### Issue Labels

improvement to existing feature, new optional setting

### Additional Information

There is a color pallet for gruvbox here: https://github.com/morhetz/gruvbox

examples of various gruvbox themes as a reference:
https://github.com/Fausto-Korpsvart/Gruvbox-GTK-Theme
https://github.com/sachnr/gruvbox-kvantum-themes
https://github.com/SylEleuth/gruvbox-plus-icon-pack

### PR DESCRIPTION
**Pull request by DeaDvey**
#  Add gruvbox base theme and gruvbox green color theme

## Pull Request Type

- [x] Feature Implementation

## Related issue
Closes #3987

## Description
I added a gruvbox base theme using the gruvbox colour palette as well as a gruvbox green color theme, I will add more of the color themes if the pr is accepted 

## Screenshots 
settings page
subscriptions page
watch page

## Testing 
I tested it and I cannot see any issues with the color schemes showing though I have not added all the translations as I only speak english.

## Desktop

- **OS:** GNU/Linux
- **OS Version:** openSUSE Tumbleweed
- **FreeTube version:** most up to date one on github

## Additional context
I wanted a gruvbox theme for FreeTube and I saw that another issue request did too, it's fine if you (the developers) don't, I was just opening this pr incase it was appropriate.

### CODE REVIEW

**Comment by DeaDvey** [comment]
Ah, lint.

**Review by absidue — COMMENTED** [review]
Thank you for opening your first pull request on the FreeTube repository 🥳.

Please fix the linter errors, if you switch to the files changed tab, you can see all the alerts.

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — CHANGES_REQUESTED** [review]
Stuff disappears when both primary and secondary colors are set to `gruvbox green`

https://github.com/FreeTubeApp/FreeTube/assets/73130443/5c92bc68-9804-4f87-84e2-35fd05c8a225

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Also this doesnt fully close the issue because you didnt implement all the gruvbox colors listed in their repo https://github.com/morhetz/gruvbox. The issue also explicitly asks for the light and dark theme implementation.

**Comment by DeaDvey** [comment]
Yes, I will implement both light and dark, I was just checking if this pr is appropriate. I'll fix the secondary colour issue right now,
Thanks

**Comment by DeaDvey** [comment]
Ok, so the reason the secondary colours were not working properaly was because I did not add the "--accent-color-opacity1 - 4" so I added that and I think it's fixed though I haven't added the gruv light theme yet.

**Comment by DeaDvey** [comment]
By the way, I'm not sure how to make the icons for it so mine were a bit thrown together and look not great, so yeah.

**Comment by DeaDvey** [comment]
image
image
image

So I added all the colour themes as well @efb4f5ff-1298-471a-8973-3d47447115dc
I tested the themes and cannot see any issues (I fixed the secondary colour scheme issue you raised)
Though I still have not added translations and the logos look bad as I don't know how to make them for the theme. 
image

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> So I added all the colour themes as well @efb4f5ff-1298-471a-8973-3d47447115dc I tested the themes and cannot see any issues (I fixed the secondary colour scheme issue you raised) Though I still have not added translations 

Can confirm that the issue is fixed and cant find any other issues at the moment :) 
You dont have to add the translations for `en_Gb` as the translators will do that for you.  

> the logos look bad as I don't know how to make them for the theme. image

Maybe previous contributors can give you some guidance on that. Summoning previous contributors that created PR's for adding themes:
@codysechelski
@Rekit0
@JonahFoster

**Comment by JonahFoster** [comment]
@DeaDvey For the logos, what I did was upload it to [Photopea ](https://www.photopea.com/)and just replace the colors on the main logo. You can find a ton of videos and blogs online about photopea, and here are some [Reddit comments](https://www.reddit.com/r/photopea/comments/bzfoem/replacing_colours_in_an_image/) that might be helpful. Feel free to reach out if you have any specific questions

**Comment by DeaDvey** [comment]
Thanks, I'll check out Photopea, I think gimp worked for me here though, for future note, I used the colorise tool to change the colour of the whole thing to #ebdbb2 and #282828

**Comment by DeaDvey** [comment]
And I realised the en_GB was not needed, I added that first as that's the one I thought to add first (I'm british) but realised en-US would be better.

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — APPROVED** [review]

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
When #4898 will be merged you have some higher quality icons available. I would recommend to redo your icons based on those enhanced ones

**Comment by DeaDvey** [comment]
Not ready right now, one of the logos is showing as dark in dark mode and the main and secondary themes don't seem to be showing, can't work on it until tomorrow though.  I did implement SVG images though which seem to work.

**Comment by DeaDvey** [comment]
Looks good to me, can't see any issues with the svgs loading and they're the correct ones.

File: src/renderer/themes.css [hunk 1]
```diff
@@ -261,6 +271,45 @@ it can be safely elided. This looks quite pleasant on this theme. */
...
+.gruvboxDark {
+  --primary-text-color: #ebdbb2;
+  --secondary-text-color: #fbf1c7;
+  --tertiary-text-color: #ebdbb2;
+  --title-color: #ebdbb2;
+  --bg-color: #1d2021;
+  --favorite-icon-color: #fabd2f;
+  --card-bg-color: #282828;
+  --secondary-card-bg-color: #282828;
+  --scrollbar-color: #32302f;
```
Comment (jasonhenriquez) [inline src/renderer/themes.css:283]: **issue (blocking):** Please choose a color that has [an acceptable color contrast](https://webaim.org/resources/contrastchecker/) (>=4.5:1) with the background (`--bg-color`). This scrollbar can be hard to find even for non-colorblind users.

File: src/renderer/themes.css [hunk 2]
```diff
@@ -261,6 +271,45 @@ it can be safely elided. This looks quite pleasant on this theme. */
...
 
+.gruvboxDark {
+  --primary-text-color: #ebdbb2;
+  --secondary-text-color: #fbf1c7;
+  --tertiary-text-color: #ebdbb2;
+  --title-color: #ebdbb2;
+  --bg-color: #1d2021;
+  --favorite-icon-color: #fabd2f;
+  --card-bg-color: #282828;
+  --secondary-card-bg-color: #282828;
```
Comment (jasonhenriquez) [inline src/renderer/themes.css:282]: **issue (blocking):** Please choose a color that has an acceptable color contrast (>=3:1) with the card background color (`--card-bg-color`). Examples of where this is used: the "About" tags on a channel, the author name outline on a comment made by the video author in a video's comments section.

**Review by jasonhenriquez — CHANGES_REQUESTED** [review]
**issue (blocking):** You'll also need to modify the [ft-share-button SCSS](https://github.com/jasonhenriquez/FreeTube/blob/development/src/renderer/components/ft-share-button/ft-share-button.scss) (note to team: we should honestly centralize this in themes.css in the future as this is very easy to miss).

**Comment by DeaDvey** [comment]
Hi, I fixed the colour contrast issues mentioned but I wasn't 100% sure on the ft-share-button thing, I think I did what you requested but you'll have to check.
I also noticed a conflicting file so updated mine to incorporate the changes but I don't have write access to resolve the conflict.

File: src/renderer/components/ft-share-button/ft-share-button.scss [hunk 3]
```diff
@@ -53,6 +54,7 @@
 
       .pastelPink &,
       .light &,
+      .gruvboxLIght &,
```
Comment (jasonhenriquez) [inline src/renderer/components/ft-share-button/ft-share-button.scss:57]:
```suggestion
      .gruvboxLight &,
```

**Comment by jasonhenriquez** [comment]
You can resolve the conflict with a merge or rebase with the `development` branch.

**Comment by jasonhenriquez** [comment]
> Hi, I fixed the colour contrast issues mentioned but I wasn't 100% sure on the ft-share-button thing, I think I did what you requested but you'll have to check.

What this governs is what version of the Invidious and YouTube logos show when you click the "Share" icon under a video. If the logo is too dark / bright to see, it means it needs to go in with either the dark or light colors. I haven't checked on this yet, but from a brief scan at the file, it seems like you got it.

**Comment by jasonhenriquez** [comment]
From what I can see, yes! As a reminder in case my message above was missed:
> You can resolve the conflict with a merge or rebase with the development branch.

**Comment by DeaDvey** [comment]
Hmmm, Lint says there's no issues on my machine.

**Review by absidue — CHANGES_REQUESTED** [review]
Please remove the `package-lock.json` file (FreeTube uses yarn) and undo the addition of the `lint` dependency.

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — DISMISSED** [review]

**Comment by jasonhenriquez** [comment]
@DeaDvey Could you please rebase/merge with development one more time and remove the changes to `yarn.lock`?

**Comment by DeaDvey** [comment]
Sorry that I took a while, I'm quite busy at the moment.

**Comment by DeaDvey** [comment]
I get an error saying there is no portal-vue. in node_modules

**Comment by absidue** [comment]
Run `yarn install` to download the dependency that you are missing.

**Comment by DeaDvey** [comment]
Oh yeah, thanks that worked, the themes all show correctly, solarized looks great too! 👍

**Review by efb4f5ff-1298-471a-8973-3d47447115dc — CHANGES_REQUESTED** [review]
Fix poo contrast issues:

- https://coolors.co/contrast-checker/fb4934-282828
- https://coolors.co/contrast-checker/b8bb26-fbf1c7
- https://coolors.co/contrast-checker/fb4934-fbf1c7
- https://coolors.co/contrast-checker/fabd2f-fbf1c7
- https://coolors.co/contrast-checker/83a598-fbf1c7
- https://coolors.co/contrast-checker/d3869b-fbf1c7
- https://coolors.co/contrast-checker/8ec07c-fbf1c7
- https://coolors.co/contrast-checker/fe8019-fbf1c7

**Comment by jasonhenriquez** [comment]
Thanks for identifying these @efb4f5ff-1298-471a-8973-3d47447115dc. Seems like `#fb1c7` should be changed to something else with a higher color contrast, from the looks of it. That and maybe having a separate `bg-color` and `card-bg-color` for `gruvboxLight`.

**Comment by jasonhenriquez** [comment]
Hi @DeaDvey, just checking in to make sure you're alright. I want to make sure the required suggestion and how to implement & validate it are clear enough.

**Comment by DeaDvey** [comment]
Yeah, I'm a bit busy right now but I should be able to have a look at it tomorrow.

**Comment by DeaDvey** [comment]
I think in order to fix this I'd either need to add a  whole new set of Color Themes for Gruvbox Light as they use different greens, reds, yellows etc or remove the Gruvbox Light base theme all together. As the ones I implemented  only have a good colour contrast with Gruvbox Dark themes.

https://github.com/morhetz/gruvbox?tab=readme-ov-file

**Comment by DontBlameMe99** [comment]
@DeaDvey Couldn't you just create gruvbox light green / gruvbox dark green respectively? Or would this be too much of a spam of colors?

**Comment by DeaDvey** [comment]
> @DeaDvey Couldn't you just create gruvbox light green / gruvbox dark green respectively? Or would this be too much of a spam of colors?

Yeah, that's what I was thinking, it would be too many colours for the menu to look nice, I could definitley do it but I don't think it's a good idea.

**Comment by jasonhenriquez** [comment]
Hi @DeaDvey, I agree that either removing the new primary/secondary color themes or removing one of the base themes (and adjusting the new primary/secondary colors in the case that Gruvbox Light is the one that's kept) should work.

**Comment by DeaDvey** [comment]
Ok thanks, jasonhenriquez, I will remove Gruvbox Light as light themes are typically less popular so I think it's thats right choice, this should fix all the mentioned contrast issues.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> Fix poor contrast issues:
> 
> - https://coolors.co/contrast-checker/fb4934-282828
> - https://coolors.co/contrast-checker/b8bb26-fbf1c7
> - https://coolors.co/contrast-checker/fb4934-fbf1c7
> - https://coolors.co/contrast-checker/fabd2f-fbf1c7
> - https://coolors.co/contrast-checker/83a598-fbf1c7
> - https://coolors.co/contrast-checker/d3869b-fbf1c7
> - https://coolors.co/contrast-checker/8ec07c-fbf1c7
> - https://coolors.co/contrast-checker/fe8019-fbf1c7
> 
> @jasonhenriquez do you have suggestions to handle this in the best possible way?

It doesnt fix the first issue in the list

**Comment by DeaDvey** [comment]
> It doesnt fix the first issue in the list
Sorry, yeah, I changed the central pane to be #1d2021 so the contrast with #fb4934 is 4.77 (good) https://coolors.co/contrast-checker/fb4934-1d2021

File: static/locales/en_GB.yaml [hunk 4]
```diff
@@ -288,6 +288,7 @@ Settings:
       Pastel Pink: Pastel pink
       Hot Pink: Hot pink
       Nordic: Nordic
+      Gruvbox: Gruvbox
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline static/locales/en_GB.yaml:291]: This should state the same as https://github.com/FreeTubeApp/FreeTube/pull/4805/files#diff-4eaadcc25bfbd7ab1ba912e36d5e1bcbab346cab59546a970df7546d5fb7a145R331

```suggestion
      Gruvbox Dark: Gruvbox Dark
```

File: src/renderer/themes.css [hunk 5]
```diff
@@ -461,6 +541,8 @@ it can be safely elided. This looks quite pleasant on this theme. */
   --logo-text-bar-color: url("../../_icons/textDraculaLightSmall.svg");
 }
 
+
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline src/renderer/themes.css:544]:
```suggestion
```

File: src/renderer/themes.css [hunk 6]
```diff
@@ -461,6 +541,8 @@ it can be safely elided. This looks quite pleasant on this theme. */
...
+
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline src/renderer/themes.css:545]:
```suggestion
```

File: src/renderer/themes.css [hunk 7]
```diff
@@ -444,6 +468,62 @@ it can be safely elided. This looks quite pleasant on this theme. */
...
+    --primary-color-active: #427b58;
+}
+
+.mainGruvboxOrange {
+    --primary-color: #fe8019;
+    --primary-color-hover: #d65d0e;
+    --primary-color-active: #af3a03;
+}
+
+
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline src/renderer/themes.css:526]:
```suggestion
```

File: src/renderer/components/theme-settings/theme-settings.js [hunk 8]
```diff
@@ -39,6 +39,7 @@ export default defineComponent({
         'black',
         // Second group
         'nordic',
+        'gruvboxDark',
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline src/renderer/components/theme-settings/theme-settings.js:42]: Please insert this in the third group below `dracula`
https://github.com/FreeTubeApp/FreeTube/pull/4805/files#diff-a0480aa13261014e61fefded30097805ea7559cfe7b068b66b209ae896b17adaR47
```suggestion
```

File: src/renderer/components/theme-settings/theme-settings.js [hunk 9]
```diff
@@ -111,6 +112,7 @@ export default defineComponent({
         this.$t('Settings.Theme Settings.Base Theme.Black'),
         // Second group
         this.$t('Settings.Theme Settings.Base Theme.Nordic'),
+        this.$t('Settings.Theme Settings.Base Theme.Gruvbox Dark'),
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline src/renderer/components/theme-settings/theme-settings.js:115]: Insert this below 
https://github.com/FreeTubeApp/FreeTube/pull/4805/files#diff-a0480aa13261014e61fefded30097805ea7559cfe7b068b66b209ae896b17adaR120
```suggestion
```

File: src/renderer/helpers/colors.js [hunk 10]
```diff
@@ -31,6 +31,13 @@ export const colors = [
   { name: 'CatppuccinMochaSapphire', value: '#74C7EC' },
   { name: 'CatppuccinMochaBlue', value: '#89B4FA' },
   { name: 'CatppuccinMochaLavender', value: '#B4BEFE' },
+  { name: 'GruvboxGreen', value: '#b8bb26' },
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline src/renderer/helpers/colors.js:34]: put this below the dracula ones

File: src/renderer/helpers/colors.js [hunk 11]
```diff
@@ -80,6 +87,13 @@ export function getColorTranslations() {
     i18n.t('Settings.Theme Settings.Main Color Theme.Catppuccin Mocha Sapphire'),
     i18n.t('Settings.Theme Settings.Main Color Theme.Catppuccin Mocha Blue'),
     i18n.t('Settings.Theme Settings.Main Color Theme.Catppuccin Mocha Lavender'),
+    i18n.t('Settings.Theme Settings.Main Color Theme.Gruvbox Green'),
```
Comment (efb4f5ff-1298-471a-8973-3d47447115dc) [inline src/renderer/helpers/colors.js:90]: put these colors below the dracula ones

File: src/main/index.js [hunk 12]
```diff
@@ -626,6 +626,8 @@ function runApp() {
           return '#de1c85'
         case 'nordic':
           return '#2b2f3a'
+        case 'gruvbox-dark':
+          return '282828'
```
Comment (jasonhenriquez) [inline src/main/index.js:630]: This needs to be updated to the new bg-color of `32302f`.

File: src/renderer/components/ft-share-button/ft-share-button.scss [hunk 13]
```diff
@@ -56,6 +57,7 @@
 
       .pastelPink &,
       .light &,
+      .gruvboxLight &,
```
Comment (jasonhenriquez) [inline src/renderer/components/ft-share-button/ft-share-button.scss:60]:
```suggestion
      .light &,
```

Remove extraneous reference to gruvBoxLight

File: src/renderer/components/ft-share-button/ft-share-button.scss [hunk 14]
```diff
@@ -79,13 +81,15 @@
...
         .hotPink &,
+	.gruvboxDark &,
         .solarizedDark &,
         .system[data-system-theme*='dark'] & {
           background-image: url('../../assets/img/invidious-logo-dark.svg');
         }
 
         .pastelPink &,
         .light &,
+	.gruvboxLight &,
```
Comment (jasonhenriquez) [inline src/renderer/components/ft-share-button/ft-share-button.scss:92]:
```suggestion
        .light &,
```

Remove extraneous reference to gruvBoxLight

File: src/renderer/themes.css [hunk 15]
```diff
@@ -444,6 +468,61 @@ it can be safely elided. This looks quite pleasant on this theme. */
...
 }
 
+.mainGruvboxGreen,
+.mainGruvboxRed,
+.mainGruvboxYellow,
+.mainGruvboxBlue,
+.mainGruvboxPurple,
+.mainGruvboxAqua,
+.mainGruvboxOrange {
+    --text-with-main-color: #282828;
```
Comment (jasonhenriquez) [inline src/renderer/themes.css:478]: This `text-with-main-color`, and the other `--text-with-accent-color`, (which should also both be the same color) are still at a notably low color contrast with `mainGruvboxRed`  and `secGruvboxRed`(4.29:1 and 4.77:1 respectively). Ideally this should be a 7:1 contrast to meet [the WCAG AAA standard](https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html).

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@DeaDvey some code reviews are still unresolved/not implemented

**Pull request closed without merging**